### PR TITLE
add query feature for GET method at api/articles end point

### PR DESCRIPTION
**Pull request by NuclearEngineer1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -55,6 +55,57 @@ describe("GET /api/articles", () => {
...
+    test("returns 400 for invalid query", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&sort_by=temperament&order=sideways")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('bad request')
+
+        });
+    });
+      test("returns [] for topic does not exist", () => {
```
Comment (MRombi) [inline __tests__/app.test.js]: I understand the reason behind the choice, but for a non existent topic we should return a 404 not found 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -55,6 +55,57 @@ describe("GET /api/articles", () => {
...
+      return request(app)
+        .get("/api/articles?topic=emus")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toEqual([])
+
+        });
+    });
```
Comment (MRombi) [inline __tests__/app.test.js]: We should add the following test cases:
- sort_by =banana
- topic is valid but does not have an article i.e. paper and responds with an empty array status 200

**Review by MRombi — CHANGES_REQUESTED** [review]
Well done on this ticket, we are almost there! Please, action on the feedback and resubmit 🙂

**Review by RichardNeat — COMMENTED** [review]
Great stuff - merge away :)

**Pull request merged**